aict_tools/scripts/apply_separation_model.py

```diff
@@ -63,35 +69,44 @@ class name given in the config file.
         feature_generation_config=model_config.feature_generation
     )
 
-    if config.has_multiple_telescopes:
+    if config.data_format == 'CTA':
         chunked_frames = []
 
-    table = config.telescope_events_key
     for df_data, start, stop in tqdm(df_generator):
-
         prediction = predict_separator(df_data[model_config.features], model)
-
-        if config.has_multiple_telescopes:
-            d = df_data[['run_id', 'array_event_id']].copy()
+        if config.data_format == 'CTA':
+            d = df_data[['obs_id', 'event_id']].copy()
             d[prediction_column_name] = prediction
             chunked_frames.append(d)
-
-        append_column_to_hdf5(data_path, prediction, table, prediction_column_name)
+            for tel in df_data['tel_id'].unique():
```

Comment:
this is a groupby operation. use `for tel_id, group in `df_data.groupby('tel_id')`